Assets/Scripts/Evidence/EvidenceDictionary.cs

```
@@ -0,0 +1,141 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+[CreateAssetMenu(fileName = "New Evidence Dictionary", menuName = "Evidence/Evidence Dictionary")]
+public class EvidenceDictionary : ScriptableObject
+{
+    [SerializeField, Tooltip("Add all the evidence required for evidence dictionary here.")]
+    private List<Evidence> _evidenceList;
+
+    private Dictionary<string, Evidence> _evidenceDictionary;
+
+    public int Count => _evidenceDictionary.Count;
+
+    /// <summary>
+    /// Converts the list of evidence objects into a dictionary so they can be accessed by name.
+    /// </summary>
+    private void OnEnable()
+    {
+        if (_evidenceList != null && (_evidenceDictionary == null || _evidenceDictionary.Count != _evidenceList.Count))
+        {
+            _evidenceDictionary = new Dictionary<string, Evidence>();
+
+            foreach (var evidence in _evidenceList)
+            {
+                AddEvidence(evidence);
+            }
+        }
+    }
+
+    /// <summary>
+    /// Replaces an Evidence object with its designated alternate evidence.
+    /// </summary>
+    /// <param name="evidence">The name of the evidence to substitute.</param>
+    public void SubstituteEvidenceWithAlt(string evidence)
+    {
+        if (!CheckIfEvidenceInDictionary(evidence))
+        {
+            return;
+        }
+
+        if (_evidenceDictionary[evidence].AltEvidence != null)
+        {
+            _evidenceDictionary[evidence] = _evidenceDictionary[evidence].AltEvidence;
+        }
```

Maybe _also_ log an error if the AltEvidence was null. Better than failing silently.

---

Done